### ISSUE TITLE
aggregateAsArray with distinct columns

### ISSUE THREAD

**Issue by threema-danilo** [juanluispaz/ts-sql-query#125]
Let's say we have a table with messages (each has a UID), a table with message reactions, and a table with message edits.

![screenshot-20240417-180832](https://github.com/juanluispaz/ts-sql-query/assets/78751145/b51ed535-56fc-4cbf-9244-bbf92da8ea11)

Every message has a 1:0-n relation with the other two tables.

In the test database, I have the following entries:

![screenshot-20240417-181056](https://github.com/juanluispaz/ts-sql-query/assets/78751145/1c8ae481-09d3-40eb-ae3b-ae020455eeff)

Using `.aggregateAsArray()`, we can achieve a query simliar to this one:

```sql
SELECT m.uid, m.text, json_group_array(r.reaction), json_group_array(e.edited) FROM messages m
LEFT OUTER JOIN reactions r ON m.uid = r.messageUid
LEFT OUTER JOIN edits e ON m.uid = e.messageUid
GROUP BY m.uid;
```

![screenshot-20240417-181220](https://github.com/juanluispaz/ts-sql-query/assets/78751145/818a41ca-35a6-47e9-a734-2ef1721a53b5)

As you can see, there is a problem: The reactions and the edits were both duplicated, because all combinations are aggregated.

Instead, I'd like a query where both `json_group_array` calls contain a `DISTINCT` clause:

```sql
SELECT m.uid, m.text, json_group_array(DISTINCT r.reaction), json_group_array(DISTINCT e.edited) FROM messages m
LEFT OUTER JOIN reactions r ON m.uid = r.messageUid
LEFT OUTER JOIN edits e ON m.uid = e.messageUid
GROUP BY m.uid;
```

![screenshot-20240417-181432](https://github.com/juanluispaz/ts-sql-query/assets/78751145/88ff11ba-961d-408b-ba28-daeb9f5d6731)

Is this possible right now? Could it be that `DISTINCT` should always be added to queries in aggregations and that the current behavior is wrong if there are multiple joins? Or alternatively, could a second variant be added which uses `DISTINCT`?

**Comment by juanluispaz** [comment]
Hi,

Using [Queries as an inline array of values](https://ts-sql-query.readthedocs.io/en/stable/queries/aggregate-as-object-array/#queries-as-an-inline-array-of-values), you can have a similar behaviour you want, but using subquery instead of joins; but not all the time a subquery is the desired behaviour, then I added for you `aggregateAsArrayDistinct` and `aggregateAsArrayOfOneColumnDistinct` to allow aggregate as array of distinct values.

This is included in ts-sql-query 1.64.0, which is already published.

Let me know your feedback.

**Comment by threema-danilo** [comment]
@juanluispaz perfect, thanks very much! Works as intended.
